src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseResource.java

```diff
@@ -249,10 +249,10 @@ public ProgrammingExerciseResource(ProgrammingExerciseRepository programmingExer
      */
     @PutMapping("/programming-exercises")
     @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
-    public ResponseEntity<ProgrammingExercise> updateProgrammingExerciseLink(@RequestBody ProgrammingExercise programmingExercise) throws URISyntaxException {
```

Comment:
I renamed the methods, as they are used both for links and automatic programming exercise setups.